apps/sms/views/conversation/js/conversation.js

```diff
@@ -547,34 +562,39 @@ var ConversationView = {
   },
 
   beforeEnterThread: function conv_beforeEnterThread(args) {
-    // TODO should we implement hooks to Navigation so that Threads could
-    // get an event whenever the panel changes?
-    Threads.currentId = args.id;
+    var threadId = +args.id;
+    return MessageManager.ensureThreadRegistered(threadId).then(() => {
```

Comment:
Yeah right and that make me think that do we need a easier way to set the thread. Here is some thought:
Because we will cache the thread in the conversation view in the future, we should have a service method that can return the current thread by id and cached for view. I think that would be more straightforward then ensureThreadRegistered. But it's definitely not in this patch, wdyt is we integrate thread create/set part into getter?